src/maps/mod.rs

```diff
@@ -299,33 +299,49 @@ struct PerfEventIterator<'a> {
 
     // technically not needed but it gives us the lifetime we need to
```

Comment:
Although it wasn't technically necessary to store a reference, I think it *was* technically necessary to have some member associated with the lifetime parameter.  So I'd remove the "technically not needed but" part from the comment, and the rest is a fine reminder of what the `PhantomData` is for.